File: frontend/src/lib/api/classify.ts
/** API functions for classification endpoints. */

import type { ClassifyResponse } from "../../types/classify";
import apiClient from "./client";

/** Trigger document classification. */
export async function triggerClassify(id: string): Promise<ClassifyResponse> {
  const response = await apiClient.post(`/classify/${id}`);
  return response.data;
}

/** Override document category. */
export async function overrideCategory(
  documentId: string,
  categoryId: string,
): Promise<void> {
  await apiClient.put(`/documents/${documentId}`, {
    documentCategoryId: categoryId,
  });
}

File: backend/src/api/schemas/classify.py
"""Pydantic schemas for classification API endpoints."""

import uuid

from pydantic import BaseModel


class ClassifyResponse(BaseModel):
    """Response after classifying a document."""

    document_id: uuid.UUID
    category_id: uuid.UUID
    category_name: str
    confidence: int
    reasoning: str
    status: str = "classified"
